In [1]:
import pandas as pd

# 1. Import des données

In [2]:
df_irve = pd.read_csv(
 'https://www.data.gouv.fr/api/1/datasets/r/eb76d20a-8501-400e-b336-d85724de5435'
)

print(f"Shape : {df_irve.shape}")
df_irve.sample(5)

/tmp/ipykernel_12907/1170431219.py:1: DtypeWarning: Columns (12,18,19,20,21,22,24,25,29,33) have mixed types. Specify dtype option on import or set low_memory=False.
  df_irve = pd.read_csv(


Shape : (213699, 52)


,nom_amenageur,siren_amenageur,contact_amenageur,nom_operateur,contact_operateur,telephone_operateur,nom_enseigne,id_station_itinerance,id_station_local,nom_station,...,datagouv_resource_id,datagouv_organization_or_owner,created_at,consolidated_longitude,consolidated_latitude,consolidated_code_postal,consolidated_commune,consolidated_is_lon_lat_correct,consolidated_is_code_insee_verified,consolidated_is_code_insee_modified
207719,QUARTZ PROPERTIES,484836218.0,hugues.bousquet@proudreed.com,WAAT Umbrella-Copro,exploitation@waat.fr,tel:+33-1-23-45-67-89,WAAT - PROUDREED,FRWA2P632011,FR*WA2*632011,PROUDREED_DARDILLY_PAISY_DC,...,8bb0a6e2-1016-42ba-aaee-f72f55c82e9f,qualicharge,2025-05-05T13:28:02.361000+00:00,4.768712,45.806182,69570.0,Dardilly,True,True,False
29046,DRIVECO,NaN,support@driveco.com,DRIVECO Partner Network,support@driveco.com,NaN,DRIVECO,FRE11PLMALBI810001,NaN,Leroy Merlin - Albi - powered by DRIVECO,...,775dd5a9-c0e4-4bb7-8995-f4b5a4148836,driveco,2026-01-30T10:47:39.688000+00:00,2.177239,43.937400,81000.0,Albi,True,True,False
65107,Maison de Santé Oasis,NaN,NaN,Freshmile | FR*FR1,roaming@freshmile.com,NaN,Freshmile France,FRFR1P3932910267840234264,1111905,Freshmile France/LLW4Q91XZ2QOOI,...,61387a4e-22f7-4662-b241-d5cac4dd91fd,gireve-2,2023-03-24T14:32:54.036000+00:00,55.478386,-20.902757,NaN,NaN,False,False,False
66052,Citivia Mulhouse,NaN,NaN,Freshmile | FR*FR1,roaming@freshmile.com,NaN,Freshmile France,FRFR1P3517829274492559064,1309357,Freshmile France/LLYMYL8UD2F9UH,...,61387a4e-22f7-4662-b241-d5cac4dd91fd,gireve-2,2023-03-24T14:32:54.036000+00:00,7.346582,47.744656,NaN,NaN,False,False,False
157924,Modulo Energies,841376734.0,support@modulo-energies.fr,Modulo,support@modulo-energies.fr,+(33)-(800)-373740,MODULO - NAZELLES NEGRON - Pl. des patis,FRS37E220174,FRS37E220174,MODULO - NAZELLES NEGRON - Pl. des patis,...,b11113db-875d-41c7-8673-0cf8ad43e917,eco-movement,2023-06-28T11:46:08.539000+00:00,0.954043,47.431091,NaN,NaN,False,False,False


Ce premier jeu de données est la base nationale des IRVE (Infrastructures de Recharge pour Véhicules Électriques).
Cette table comporte 212234 lignes pour 52 variables.

In [3]:
df_ve = pd.read_csv(
 'https://www.data.gouv.fr/api/1/datasets/r/90e0d717-deda-4bdc-9987-f82faac5bc93',
 encoding='latin-1'
)

print(f"Shape : {df_ve.shape}")
df_ve.sample(5)

Shape : (703545, 8)


/tmp/ipykernel_12907/2115914144.py:1: DtypeWarning: Columns (0,2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_ve = pd.read_csv(


,CODGEO,LIBGEO,EPCI,LIBEPCI,DATE_ARRETE,NB_VP_RECHARGEABLES_EL,NB_VP_RECHARGEABLES_GAZ,NB_VP
100390,81243,SAINT-BEAUZILE,200066124.0,CA Gaillac-Graulhet,2021-03-31,0,0,176
694728,87052,LA CROIX-SUR-GARTEMPE,200071942.0,CC Haut Limousin en Marche,2022-12-31,1,0,248
158821,44999,FORAINS,NaN,NaN,2021-09-30,0,0,1227
131194,22195,PLEUBIAN,200065928.0,CA Lannion-Trégor Communauté,2024-03-31,46,0,2939
89137,72046,LE BREIL-SUR-MÉRIZE,200072684.0,CC Le Gesnois Bilurien,2024-06-30,25,0,1752


Ce second jeu de données contient des informations sur les voitures particulières immatriculées par commune et par type de recharge. Il contient l'historique des immatriculations pour chaque commune à différentes dates.
Cette table comporte 703545 lignes pour 8 variables. 

In [4]:
df_revenus = pd.read_csv(
 'https://www.data.gouv.fr/api/1/datasets/r/516130bc-4dcb-47f5-8347-ae96553c43ab',sep=';'
)

print(f"Shape : {df_revenus.shape}")
df_revenus.sample(5)

Shape : (34926, 57)


/tmp/ipykernel_12907/3484954150.py:1: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  df_revenus = pd.read_csv(


,Nom géographique GMS,Code géographique,Libellé géographique,[DISP] Nbre de ménages fiscaux,[DISP] Nbre de personnes dans les ménages fiscaux,[DISP] Nbre d'unités de consommation dans les ménages fiscaux,[DISP] 1ᵉʳ quartile (€),[DISP] Médiane (€),[DISP] 3ᵉ quartile (€),[DISP] Écart interquartile (€),...,[DEC] 9ᵉ décile (€),[DEC] Rapport interdécile D9/D1,[DEC] S80/S20,[DEC] Iice de Gini,[DEC] Part des revenus d’activité (%),[DEC] dont part des salaires et traitements (%),[DEC] dont part des iemnités de chômage (%),[DEC] dont part des revenus des activités non salariées (%),"[DEC] Part des pensions, retraites et rentes (%)",[DEC] Part des autres revenus (%)
32217,Lizac,82099,Lizac,212.0,525.0,351.9,NaN,21190.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
32569,Rustrel,84103,Rustrel,295.0,613.0,438.9,NaN,20920.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
9772,Sainte-Colombe-la-Commanderie,27524,Sainte-Colombe-la-Commaerie,312.0,806.0,526.6,NaN,24580.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
17637,Mûrs-Erigné,49223,Mûrs-Erigné,2727.0,6258.0,4283.5,19210.0,24300.0,31020.0,11820.0,...,44540.0,"3,7",5.1,0.310,65.0,57.6,2.3,5.1,29.6,5.4
34816,Persan,95487,Persan,4828.0,12708.0,8159.4,14260.0,19590.0,25070.0,10810.0,...,32310.0,"5,3",6.5,0.312,81.5,76.2,4.2,1.1,17.6,0.9


Ce dernier jeu de données contient des informations sur les voitures particulières immatriculées par commune et par type de recharge.
Cette table comporte 34926 lignes pour 57 variables.

# 2. Jointure des tables

## 2.1 Choix de la variable de jointure

Les 3 jeu de données contiennent une variable indiquant le code géographique mais sous 3 noms différents :
- `code_insee_commune` pour df_irve
- `CODGEO` pour df_ve
- `Code géographique` pour df_revenus

La jointure sera faite sur ces variables.

## 2.2 Préparation de la variable de jointure

Le diagnostic met en évidence plusieurs points d’attention :
- Présence de valeurs manquantes dans `code_insee_commune` (df_irve)  
- Présence de longueurs de codes incohérentes :
  - `code_insee_commune` (df_irve) contient des codes de longueur 3, 6 et 7
  - `CODGEO` (df_ve) contient des codes de longueur 4
- Les variables de jointure ne sont pas uniques :
  - `code_insee_commune` (df_irve)
  - `CODGEO` (df_ve)

Les clés de jointure présentent des problèmes de qualité et de format.  
Il est nécessaire de les nettoyer et les uniformiser (type, format, gestion des valeurs manquantes, unicité) avant d’effectuer la jointure.

### 2.2.1 Unification des codes

In [5]:
from src.preparation_data import nettoyer_code_insee

In [6]:
df_irve["code_insee_commune"] = df_irve["code_insee_commune"].apply(nettoyer_code_insee)
df_ve["CODGEO"] = df_ve["CODGEO"].apply(nettoyer_code_insee)
df_revenus["Code géographique"] = df_revenus["Code géographique"].apply(nettoyer_code_insee)

Les codes insee sont maintenant au même format dans les 3 bases de données : un code à 5 caractères.

### 2.2.2 Compléter les valeurs manquantes

In [7]:
from src.preparation_data import creer_gdf_irve, charger_communes, joindre_communes, ajouter_codes_geo

In [8]:
gdf_irve = creer_gdf_irve(df_irve, "consolidated_longitude", "consolidated_latitude")
gdf_communes = charger_communes()
gdf_result = joindre_communes(gdf_irve, gdf_communes)
df_irve = ajouter_codes_geo(df_irve, gdf_result)

There was an error while reading the file from the URL: https://minio.lab.sspcloud.fr/projet-cartiflette/production/provider=IGN/dataset_family=ADMINEXPRESS/source=EXPRESS-COG-CARTO-TERRITOIRE/year=2022/administrative_level=COMMUNE/crs=4326/DEPARTEMENT=20/vectorfile_format=geojson/territory=metropole/simplification=0/raw.geojson
Error message: '/vsimem/pyogrio_deeeffcabd1243998ab712996ea34aca' not recognized as being in a supported file format.; It might help to specify the correct driver explicitly by prefixing the file path with '<DRIVER>:', e.g. 'CSV:path'.


Nous avons maintenant 2 colonnes concernant le code géographique dans la base de données IRVE :
- `code_insee_commune` : la variable initiale
- `code_geo_total` : la variable recalculant tous les codes géographiques à partir de la latitude et la longitude

In [9]:
from src.preparation_data import compter_valeurs_manquantes, compter_uniques

In [10]:
colonnes_irve = ["code_insee_commune", "code_geo_total"]

print("Valeurs manquantes :")
print(compter_valeurs_manquantes(df_irve, colonnes_irve))

print("\nValeurs uniques :")
print(compter_uniques(df_irve, colonnes_irve))

Valeurs manquantes :
{'code_insee_commune': np.int64(64966), 'code_geo_total': np.int64(1818)}

Valeurs uniques :
{'code_insee_commune': 9440, 'code_geo_total': 10278}


Cette manipulation permet de passer de 57919 valeurs manquantes à 1768. De plus les codes recalculés sont plus fiables que ceux initiaux. En effet, les données entrées n'étant pas toujours vérifiées, des erreurs étaient présentes. Certaines code géographiques correspondaient au code postal par exemple.

## 2.3 Jointure

Réfléchissons aux variables à garder... Lesquelles pourraient être intéressante ?

In [11]:
list(df_irve.columns)

['nom_amenageur',
 'siren_amenageur',
 'contact_amenageur',
 'nom_operateur',
 'contact_operateur',
 'telephone_operateur',
 'nom_enseigne',
 'id_station_itinerance',
 'id_station_local',
 'nom_station',
 'implantation_station',
 'adresse_station',
 'code_insee_commune',
 'coordonneesXY',
 'nbre_pdc',
 'id_pdc_itinerance',
 'id_pdc_local',
 'puissance_nominale',
 'prise_type_ef',
 'prise_type_2',
 'prise_type_combo_ccs',
 'prise_type_chademo',
 'prise_type_autre',
 'gratuit',
 'paiement_acte',
 'paiement_cb',
 'paiement_autre',
 'tarification',
 'condition_acces',
 'reservation',
 'horaires',
 'accessibilite_pmr',
 'restriction_gabarit',
 'station_deux_roues',
 'raccordement',
 'num_pdl',
 'date_mise_en_service',
 'observations',
 'date_maj',
 'cable_t2_attache',
 'last_modified',
 'datagouv_dataset_id',
 'datagouv_resource_id',
 'datagouv_organization_or_owner',
 'created_at',
 'consolidated_longitude',
 'consolidated_latitude',
 'consolidated_code_postal',
 'consolidated_commune',
 '

In [12]:
list(df_ve.columns)

['CODGEO',
 'LIBGEO',
 'EPCI',
 'LIBEPCI',
 'DATE_ARRETE',
 'NB_VP_RECHARGEABLES_EL',
 'NB_VP_RECHARGEABLES_GAZ',
 'NB_VP']

In [13]:
list(df_revenus.columns)

['Nom géographique GMS',
 'Code géographique',
 'Libellé géographique',
 '[DISP] Nbre de ménages fiscaux',
 '[DISP] Nbre de personnes dans les ménages fiscaux',
 "[DISP] Nbre d'unités de consommation dans les ménages fiscaux",
 '[DISP] 1ᵉʳ quartile (€)',
 '[DISP] Médiane (€)',
 '[DISP] 3ᵉ quartile (€)',
 '[DISP] Écart interquartile (€)',
 '[DISP] 1ᵉʳ décile (€)',
 '[DISP] 2ᵉ décile (€)',
 '[DISP]3ᵉ décile (€)',
 '[DISP] 4ᵉ décile (€)',
 '[DISP] 6ᵉ décile (€)',
 '[DISP] 7ᵉ décile (€)',
 '[DISP] 8ᵉ décile (€)',
 '[DISP] 9ᵉ décile (€)',
 '[DISP] Rapport interdécile 9ᵉ décile/1ᵉʳ decile',
 '[DISP] S80/20',
 '[DISP] Iice de Gini',
 '[DISP] Part des revenus d’activité (%)',
 '[DISP] dont part des salaires et traitements(%)',
 '[DISP] dont part des iemnités de chômage (%)',
 '[DISP] dont part des revenus des activités non salariées (%)',
 '[DISP] Part des pensions, retraites et rentes (%)',
 '[DISP] Part des revenus du patrimoine et autres revenus (%)',
 "[DISP] Part de l'ensemble des prestat

### 2.3.1 Une ligne par code géo

Pour pouvoir efectuer la jointure sur les codes géographiques, il est essentiel que pour chacune de nos 3 bases de données, chaque ligne corresponde à un unique code géographique.
Il faut alors réfléchir à la façon dont on veut agréger df_irve et quelles variables on souhaite garder.
Il faut également étudier la base df_ve afin de supprimer les "doublons" de code géographique.

##### df_irve

Ici, l'objectif est de mesurer l'offre de recharge par commune. Comme il peut y avoir plusieurs points de recharge par commune, il faut agréger. L'objectif final est d'expliquer ou prédire le taux de véhicules électriques local.

Après étude des nos variables (cf etude_df_irve.ipynb) nous avons sélectionner plusieurs variables nous permettant d'en créer de nouvelles, agrégées par code géographique.

Certaines variables ont été écarté directement car jugées non pertinente pour notre sujet, d'autres présetaient trop de valeurs manquantes, d'autres étaient trop peu informatives ou encore était du texte de libre donc trop difficile à utiliser avec le peu de temps dont nous disposons.

variables utilisées (12 variables) pour la création des variables agrégées :`code_geo_total`,`nom_operateur`,
               'implantation_station',
               'nbre_pdc',
               'puissance_nominale',
               'prise_type_ef',
               'prise_type_2',
               'prise_type_combo_ccs',
               'prise_type_chademo',
               'gratuit',
               'paiement_cb',
               'paiement_autre'

variables créées : une ligne finale = un code géographique.

|variables utilisées| Variable finale   | Construction      |
|| ----------------- | ----------------- |
|| total_pdc         | somme             |
|| puissance_moyenne | moyenne           |
|| puissance_max     | max               |
|| nb_operateurs     | nunique           |
|| top_operateur     | mode              |
|| pct_type_2        | moyenne booléenne |
|| pct_gratuit       | moyenne booléenne |
|| part_voirie       | dummies + moyenne |

Compléter uniquement se tableau très proprement

Commençons par rendre les données des variables sélectionner propres avant de faire l'agrégation.

In [14]:
from src.nouvelles_variables import clean_irve_variables_finales

df_filtre = clean_irve_variables_finales(df_irve)
df_filtre.sample(5)

,code_geo_total,nom_operateur,implantation_station,nbre_pdc,puissance_nominale,prise_type_ef,prise_type_2,prise_type_combo_ccs,prise_type_chademo,gratuit,paiement_cb,paiement_autre,borne_rapide
22847,44184,ChargePoint Austria GmbH | FR*CG0,Parking privé à usage public,3,22.0,False,True,False,False,False,True,True,False
207248,93005,WAAT SAS | FR*WA2,Parking public,3,47.0,False,False,True,True,False,False,True,True
6208,37195,Green To Wheel,Parking privé à usage public,10,300.0,False,True,True,False,False,True,True,True
107632,59011,TotalEnergies Charging Services,Voirie,3,22.0,True,True,False,False,<NA>,False,True,False
136699,75056,GROUPE INDIGO,Parking public,140,7.4,False,True,False,False,<NA>,<NA>,<NA>,False


Nous pouvons maintenant procéder à la création des variables agrégées.

In [15]:
from src.nouvelles_variables import creer_features_irve

nv_var = creer_features_irve(df_filtre)
nv_var.sample(5)

,code_geo_total,total_pdc,puissance_moyenne,puissance_max,nb_operateurs,pct_type_2,pct_combo_ccs,pct_chademo,pct_type_ef,pct_gratuit,pct_paiement_cb,pct_paiement_autre,top_operateur,Parking privŽ rŽservŽ ˆ la clientle,Parking privŽ ˆ usage public,Parking privé réservé à la clientèle,Parking privé à usage public,Parking public,Station dédiée à la recharge rapide,Voirie
9560,86194,3158,88.316364,600.0,27,0.614625,0.351779,0.025692,0.039526,0.0,0.244898,0.656379,Soregies,0.0,0.0,0.0,0.312253,0.158103,0.039526,0.490119
723,09032,106,18.113750,46.0,4,0.96875,0.09375,0.09375,0.375,0.0,0.0,0.6875,ELECTRIC 55 CHARGING,0.0,0.0,0.0,0.0,0.25,0.0,0.75
9308,84034,24,22.000000,22.0,2,1.0,0.0,0.0,1.0,0.0,1.0,0.0,inconnu,0.0,0.0,0.0,0.0,0.0,0.0,1.0
1985,22040,4,22.000000,22.0,1,1.0,0.0,0.0,0.0,0.0,0.0,0.0,Bouygues Energies & Services,0.0,0.0,0.0,0.0,0.0,0.0,1.0
5228,50498,18,69.166667,100.0,2,0.333333,0.333333,0.333333,0.0,0.0,1.0,1.0,Zunder (Grupo Easycharger S.A),0.0,0.0,0.0,0.0,0.0,0.5,0.5


##### df_ve

La base véhicules contient plusieurs dates d'observation pour une même commune.

Afin d'être cohérent avec les autres sources de données (IRVE, INSEE), nous souhaitons disposer d'une seule ligne par commune correspondant à la situation la plus récente disponible.

Nous conservons donc, pour chaque code commune (`CODGEO`), la dernière observation temporelle.

In [16]:
from src.nouvelles_variables import preparer_base_ve

In [17]:
df_ve_latest = preparer_base_ve(df_ve)
print(df_ve_latest.shape)
df_ve_latest.head()

(35197, 5)


,CODGEO,DATE_ARRETE,NB_VP,NB_VP_RECHARGEABLES_EL,taux_equipement_ve
216883,01001,2025-09-30,968,30,0.030992
432383,01002,2025-09-30,286,5,0.017483
541297,01004,2025-09-30,16080,518,0.032214
109071,01005,2025-09-30,2306,68,0.029488
216903,01006,2025-09-30,169,4,0.023669


Deux variables sont conservées :

- `NB_VP` : nombre total de voitures particulières
- `NB_VP_RECHARGEABLES_EL` : nombre de voitures rechargeables / électriques

Plutôt que d'utiliser le nombre brut de véhicules électriques, nous construisons un ratio :

\[
taux\_equipement\_ve = \frac{NB\_VP\_RECHARGEABLES\_EL}{NB\_VP}
\]

Ce ratio est plus pertinent car il neutralise l'effet de taille des communes.

In [18]:
df_ve_latest["taux_equipement_ve"].describe().round(4)

count    35197.0000
mean         0.0229
std          0.0182
min          0.0000
25%          0.0128
50%          0.0205
75%          0.0302
max          1.0000
Name: taux_equipement_ve, dtype: float64

##### df_revenus

sous-partie à faire

In [ ]:
# On garde le code commune et le revenu médian (souvent colonne 'MED21' pour 2021)
df_revenus_final = df_revenus[['COM', 'MED21']].copy()
df_revenus_final.rename(columns={'COM': 'code_commune_insee', 'MED21': 'revenu_median'}, inplace=True)

**Variables à garder :**
[DISP] Médiane (€), [DISP] Part des revenus d’activité (%), [DISP] Iice de Gini.

Intérêt : La médiane du revenu est indispensable pour vérifier si l'adoption du VE est corrélée à la richesse de la commune. L'indice de Gini peut montrer si les communes très inégalitaires ont un comportement différent.

On peut en choisir d'autres. D'ailleurs on doit choisir uniquement les variables de df_irve avant la jointure car pour les 2 autres tables on a déjà une ligne par commune donc pas besoin de faire d'agrégation. Ainsi on peut toutes les prendre et choisir après.

In [19]:
# jointure
df_final = (
    df_ve_latest
    .merge(nv_var, left_on="CODGEO", right_on="code_geo_total", how="left")
    .merge(df_revenus, left_on="CODGEO", right_on="Code géographique", how="left")
)

print("Shape base finale :", df_final.shape)

df_final.head()

Shape base finale : (35197, 82)


,CODGEO,DATE_ARRETE,NB_VP,NB_VP_RECHARGEABLES_EL,taux_equipement_ve,code_geo_total,total_pdc,puissance_moyenne,puissance_max,nb_operateurs,...,[DEC] 9ᵉ décile (€),[DEC] Rapport interdécile D9/D1,[DEC] S80/S20,[DEC] Iice de Gini,[DEC] Part des revenus d’activité (%),[DEC] dont part des salaires et traitements (%),[DEC] dont part des iemnités de chômage (%),[DEC] dont part des revenus des activités non salariées (%),"[DEC] Part des pensions, retraites et rentes (%)",[DEC] Part des autres revenus (%)
0,01001,2025-09-30,968,30,0.030992,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,01002,2025-09-30,286,5,0.017483,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,01004,2025-09-30,16080,518,0.032214,01004,542.0,129.367750,400.0,7.0,...,39380.0,"5,6",8.1,0.368,67.9,61.5,3.2,3.2,24.4,7.7
3,01005,2025-09-30,2306,68,0.029488,01005,8.0,26.666667,50.0,2.0,...,43610.0,"3,2",4.4,0.298,73.2,67.2,2.2,3.8,19.7,7.1
4,01006,2025-09-30,169,4,0.023669,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


# 3. Analyse des variables

Après le choix des variables : 
0. variable cible = taux de VE
1. faire des analyses descriptives : 
- automatiser avec une fonction
- matrice de corrélation (heatmap)
2. modélisation :
simple regression ou plus avace (random forest...)

# 4. Modélisation

In [21]:
from src.modelisation import modelisation_lasso, modelisation_xgboost

Nos variables pour la modélisation sont les suivantes :

In [22]:
features = ['revenu_median', 'total_pdc', 'puissance_max', 'nb_operateurs', 'pct_gratuit']
target = 'taux_equipement_ve' 

# 2. Exécuter le Lasso (Baseline & Sélection)
model_l, metrics_l, coeffs_l = modelisation_lasso(df_final, features, target)
print(f"R2 Lasso : {metrics_l['r2']:.3f}")
print("Déterminants (Lasso) :\n", coeffs_l[coeffs_l != 0])

# 3. Exécuter le Gradient Boosting (Performance)
model_x, metrics_x, importance_x = modelisation_xgboost(df_final, features, target)
print(f"R2 XGBoost : {metrics_x['r2']:.3f}")
importance_x.plot(kind='barh', title="Importance des variables - XGBoost")

KeyError: "['revenu_median'] not in index"